rust_dev_preview/cross_service/detect_faces/src/main.rs

```diff
@@ -205,13 +206,13 @@ async fn main() -> Result<(), Box<dyn Error>> {
         println!();
     }
 
-    let s3_shared_config = aws_config::from_env()
```

Comment:
I think we should be pushing most folks to the Cargo feature, but up to for discussion

Reply:
Setting it to latest seems like the anti-pattern to me, at least for examples. It should be clear that this needs to be set. Although, perhaps `BehaviorMajorVersion::latest()` is the wrong one to use and we should just always pin it to the actual latest version in examples.

Reply:
From a customer standpoint, manually specifying BehaviorVersion in every single example will be generally unergonomic, and in the examples, will serve to clutter and obfuscate the educational content. So I guess there's two questions here? One, do you expect every load to go through `from_env_with_version` in the common case? Two, is it appropriate for every example to have that overhead? I'll let y'all decide on 1, but for the examples repo, I'm going to ask for the feature flag generally with a dedicated example in `sdk_config` to show the `_with_version` mechanisms.

Reply:
The decision we make here should be in sync with a code snippet that may be included when we make a GA announcement.

Reply:
The default path should be to specify a behavior version manually, with being able to not specify it (and thus default to latest) being an opt-in Cargo feature. We need it to be this way or this entire feature is completely meaningless. The point is to be able to make behavior breaks in the future without actually breaking anyone.

For example, we only partially implemented default timeouts, and we're going to need a behavior break in the future to fully implement it. That behavior break is to have a different default timeout value on mobile devices, which could break someone's application. But if they pin to a behavior version, then it won't break them, and they can upgrade it at their convenience with adequate testing.

I suggested to @rcoh that we rename this method, however. `from_env_with_version` is just terribly long and gross. I think we should have `defaults(behavior_major_version)`, and `load_defaults(behavior_major_version)`, as replacements for `from_env_with_version` and `load_from_env_with_version` respectively.

Reply:
Did a bunch of renames in https://github.com/smithy-lang/smithy-rs/pull/3204

From an offline discussion, we'll make the getting started, SDK config, and cross service examples explicitly set the behavior version. The rest of the examples will use the feature to use the latest.